app/dao/notifications_dao.py

```diff
@@ -474,6 +474,27 @@ def dao_update_notifications_sent_to_dvla(job_id, provider):
 
 
 @statsd(namespace="dao")
+@transactional
+def dao_update_notifications_by_reference(references, update_dict):
+    now = datetime.utcnow()
+    updated_count = Notification.query.filter(
+        Notification.reference.in_(references)
```

Comment:
The in_ does have a limit of how many it can take before breaking. Let's make sure this is under with the max number you will send this function.

Reply:
tested with 10,000 random references, runs in under a sec including db setup/tear down stuff